### PR TITLE
Create sync script for all devices which

### PR DESCRIPTION
**Pull request by torbacz**
## Description

"Simple" PS script to make sure every StyleCop settings (including packages, nfproj and stylecop.settings) are up to date in every project.

## How Has This Been Tested?

- Ran multiple times on AD5328 project

### CODE REVIEW

File: devices/AD5328/AD5328.nfproj [hunk 1]
```diff
@@ -80,4 +81,11 @@
     </PropertyGroup>
     <Error Condition="!Exists('packages\Nerdbank.GitVersioning.3.5.103\build\Nerdbank.GitVersioning.targets')" Text="$([System.String]::Format('$(ErrorText)', 'packages\Nerdbank.GitVersioning.3.5.103\build\Nerdbank.GitVersioning.targets'))" />
   </Target>
-</Project>
\ No newline at end of file
+  <Import Project=".\packages\StyleCop.MSBuild.6.2.0\build\StyleCop.MSBuild.targets" Condition="Exists('.\packages\StyleCop.MSBuild.6.2.0\build\StyleCop.MSBuild.targets')" />
```
Comment (josesimoes) [inline devices/AD5328/AD5328.nfproj:84]: These paths look better without starting with `.\` don't they?

Comment (torbacz) [reply devices/AD5328/AD5328.nfproj]: For me, it's the same, but changed it anyway.

Comment (josesimoes) [reply devices/AD5328/AD5328.nfproj]: Sure! Just that all the others are using the "no dot" notation and thought of making this consistent. Picky of me... 😅

**Comment by Ellerbach** [comment]
Thanks! So the script can be run on all directories? If yes, great! We will then have to run it slowly on batch because there will be few fixes on the way for sure.

**Comment by torbacz** [comment]
Yes, but you need to modify script slightly. There is a function: isProjectWhitelisted either remove it to run against all project or add new project name.

**Comment by Ellerbach** [comment]
> Yes, but you need to modify script slightly. There is a function: isProjectWhitelisted either remove it to run against all project or add new project name.

So can I ask you to write a short README to explain how to use the script? in short the different options and one or two examples: run it everywhere, run it for specific directories or equivalent?
Thanks again!

File: StyleCop/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Description
```
Comment (Ellerbach) [inline StyleCop/README.md:1]:
```suggestion
# Description

```

File: StyleCop/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+This script was created to sync Stylecop settings across all projects in pointed path. It's iterating through all folders looking for .nfproj files.
+If any found, then update it configuration to use Stylecop with settings from this folder. 
+
+# How to use
```
Comment (Ellerbach) [inline StyleCop/README.md:5]:
```suggestion
## How to use

```

File: StyleCop/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+Run this script from power shell console.
+
+``
```
Comment (Ellerbach) [inline StyleCop/README.md:8]:
```suggestion
```shell
```

File: StyleCop/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+.\syncSettingsOverEveryDevice.ps1
+``
```
Comment (Ellerbach) [inline StyleCop/README.md:10]:
```suggestion
```
```

File: StyleCop/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+# Configuration
```
Comment (Ellerbach) [inline StyleCop/README.md:12]:
```suggestion
## Configuration
```

**Review by Ellerbach — COMMENTED** [review]
Thanks! Looks good to me all up, thanks for the addition of readme, suggestion to make it properly markdown lint (yea, there is a linter as well for Markdown not *yet* in place here ;-))
Then good to merge and to use on more elements.
The rule are the ones I've adjusted, so seems good to me if you haven't change anything (but doesn't seems so).
Next step will also to document the fact we need to manually add the linter or use the script for people who wants to add a new binding.
And this will simplify a lot the reviews, 70% of the comments are liteeraly on thos elements.
Thanks again!

**Comment by josesimoes** [comment]
LGTM! Shall we merge this? 😄

**Comment by Ellerbach** [comment]
OK, so let's give it a try, we can always adjust moving forward! HUge thanks again!

**Review by josesimoes — APPROVED** [review]
Another great tool for our toolbox! 💯

**Pull request merged**